spec/policy/rate_limit/redis_shdict_spec.lua

```diff
@@ -0,0 +1,55 @@
+local env = require('resty.env')
+local ts = require ('apicast.threescale_utils')
+
+local redis_host = env.get('TEST_NGINX_REDIS_HOST') or 'localhost'
+local redis_port = env.get('TEST_NGINX_REDIS_PORT') or 6379
+
+local redis_shdict = require('apicast.policy.rate_limit.redis_shdict')
+
+describe('Redis Shared Dictionary', function()
+    local redis
+    local shdict
+
+    before_each(function()
+        local options = { host = redis_host, port = redis_port, db = 1 }
+        redis = assert(ts.connect_redis(options))
+        shdict = assert(redis_shdict.new(options))
+
+        assert(redis:flushdb())
+    end)
+
+    describe('flush_all', function()
+        it('removes all records', function()
+            assert(redis:set('foo', 'bar'))
+            assert.equal('bar', redis:get('foo'))
+
+            shdict:flush_all()
+
+            assert.equal(ngx.null, redis:get('foo'))
+        end)
+    end)
+
+    describe('incr', function()
+        pending('without default', function()
+            local ret, err = shdict:incr('somekey', 2)
+
+            -- TODO: nginx shdict:incr returns: nil, 'not found' when the key does not exist
```

Comment:
This is what the 2 assertions below are checking, right?

Reply:
Sorry, didn't notice this one was marked as 'pending'

Reply:
Yep.